### PR TITLE
fix(select): deselect old options when programmatically setting value

### LINKED ISSUE #4476: Select option styling bug
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Select component: setting an option programmatically should clear any existing previous selected item styles.

#### What is the current behavior?
The selection is changed, but the previously selected item has still the selected style

#### What are the steps to reproduce?
In the Plunker: 
- Open the select, notice the styling of "tacos", don't change anything, close the select
- click the reset button
- Open the select again: "pizza" and "tacos" have both the selected styling (mat-selected)

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
http://plnkr.co/edit/o866DpIMsaYVmPPjwNRf?p=preview

#### What is the use-case or motivation for changing an existing behavior?
consisten look

#### Which versions of Angular, Material, OS, browsers are affected?
tested it with Chrome 54

### PR DESCRIPTION
**Pull request by willshowell**
Fixes #4476

This may be naive regarding selection change events, but existing tests pass and it fixes the issue.

cc @crisbeto

### CODE REVIEW

**Comment by crisbeto** [comment]
Off the top of my head, I can't see anything that it might break, but let me play around with it a bit more tomorrow.

File: src/lib/select/select.ts [hunk 1]
```diff
@@ -544,12 +544,12 @@ export class MdSelect implements AfterContentInit, OnDestroy, OnInit, ControlVal
       throw getMdSelectNonArrayValueError();
     }
 
+    this._clearSelection();
```
Comment (crisbeto) [inline src/lib/select/select.ts]: nit: Can you add a new line after this one?

File: src/lib/select/select.spec.ts [hunk 2]
```diff
@@ -300,6 +300,35 @@ describe('MdSelect', () => {
...
+
+      control.setValue(foods[1].value);
+      fixture.detectChanges();
+
+      trigger.click();
+      fixture.detectChanges();
+
+      options =
+        overlayContainerElement.querySelectorAll('md-option') as NodeListOf<HTMLElement>;
+      expect(options[0].classList).not.toContain('mat-selected');
```
Comment (crisbeto) [inline src/lib/select/select.spec.ts]: Could you add a description to the expectations in the test? It makes it easier to parse if one of them fails. E.g. `expect(options[0].classList).not.toContain('mat-selected', 'Expected first option not to be selected');`.

**Comment by willshowell** [comment]
@crisbeto changes have been made

**Comment by crisbeto** [comment]
Seems like there are a couple of linter errors. The Closure failure should be fixed by https://github.com/angular/material2/pull/4709.

**Comment by crisbeto** [comment]
@willshowell the two PRs got merged. Can you rebase from master?

**Pull request merged**